From 2222222222222222222222222222222222222222 Mon Sep 17 00:00:00 2001
From: conda-forge <conda-forge@anaconda.com>
Date: Thu, 28 Aug 2026 12:00:00 +0000
Subject: [PATCH] Use Python 3 C API names dropped by SWIG 4.3

SWIG < 4.3 emitted Python 2 compatibility macros (PyInt_AsLong,
PyInt_FromLong, PyString_FromString) into the generated wrapper, which
M2Crypto's interface files relied on. SWIG 4.3 removed them, so these
call sites now fail to compile against Python 3. Use the Python 3 names
directly.

---
 src/SWIG/_ec.i  | 4 ++--
 src/SWIG/_lib.i | 2 +-
 src/SWIG/_rsa.i | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

--- a/src/SWIG/_ec.i
+++ b/src/SWIG/_ec.i
@@ -154,9 +154,9 @@
     PyDict_SetItemString(ret_dict, "NID",
                          PyLong_FromLong((long)curves[i].nid));
     PyDict_SetItemString(ret_dict, "sname",
-                         PyString_FromString(sname));
+                         PyUnicode_FromString(sname));
     PyDict_SetItemString(ret_dict, "comment",
-                         PyString_FromString(comment));
+                         PyUnicode_FromString(comment));
 
     PyTuple_SET_ITEM(ret_tuple, i, ret_dict);
 
--- a/src/SWIG/_lib.i
+++ b/src/SWIG/_lib.i
@@ -345,7 +345,7 @@
          */
         cret = 0;
     } else {
-        cret = (int)PyInt_AsLong(ret);
+        cret = (int)PyLong_AsLong(ret);
     }
 
     Py_XDECREF(ret);
--- a/src/SWIG/_rsa.i
+++ b/src/SWIG/_rsa.i
@@ -547,6 +547,6 @@
         return NULL;
     }
 
-    return PyInt_FromLong(*data);
+    return PyLong_FromLong(*data);
 }
 %}
